### PR TITLE
Adds settings test

### PR DESCRIPTION
**Pull request by TomKlotzPro**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | no   
| BC breaks?        | no     
| Related Issue     | https://github.com/algolia/algoliasearch-client-php/issues/527
| Need Doc update   |no

## Describe your change

This pull request is related to : https://github.com/algolia/algoliasearch-client-php/issues/527

### CODE REVIEW

**Comment by TomKlotzPro** [comment]
@nunomaduro in this file, testSettingsWithReplicas is not correct anymore, issue may come from line : 145

File: tests/Integration/SettingsTest.php [hunk 1]
```diff
@@ -19,18 +61,48 @@ protected function setUp()
...
         $retrievedSettings = $index->getSettings();
-        $this->assertArraySubset($this->settings, $retrievedSettings);
+        self::assertArraySubset($this->settings, $retrievedSettings);
     }
 
     /**
-     * @depends testSettingsCanBeUpdatedAndRetrieved
+     * @depends testSettings
      */
     public function testSettingsWithReplicas()
```
Comment (nunomaduro) [inline tests/Integration/SettingsTest.php]: You should try to understand why this test exists, and adapt it concerning the new array settings.

File: tests/Integration/SettingsTest.php [hunk 2]
```diff
@@ -19,18 +61,48 @@ protected function setUp()
...
+
+        /* Get the settings with getSettings  */
+        $retrievedSettings = $index->getSettings();
+        self::assertArraySubset($this->settings, $retrievedSettings);
+
+        /* Set the settings with the following parameters with setSettings */
+        $responses[] = $index->setSettings(array('typoTolerance' => 'min', 'ignorePlurals' => array('en', 'fr'), 'removeStopWords' => array('en', 'fr'), 'distinct' => true));
+
+        /* Wait all collected task to terminate */
+        foreach ($responses as $r) {
```
Comment (nunomaduro) [inline tests/Integration/SettingsTest.php]: You can use the multiResponses object here.

File: tests/Integration/SettingsTest.php [hunk 3]
```diff
@@ -19,18 +61,48 @@ protected function setUp()
...
-        $index->setSettings($this->settings);
+        /* Add one record to create the index with saveObject */
+        $object = array('objectID' => 1, 'name' => 'foo');
+        $index->saveObject($object);
+
+        /* Set the settings with $settings with setSettings */
+        $responses[] = $index->setSettings($this->settings);
+
+        /* Wait all collected task to terminate */
+        foreach ($responses as $r) {
```
[the remark this answers is not in the document]
Comment (TomKlotzPro) [reply tests/Integration/SettingsTest.php]: done

File: tests/Integration/SettingsTest.php (hunk 1 above)
Comment (TomKlotzPro) [reply tests/Integration/SettingsTest.php]: done

File: tests/Integration/SettingsTest.php (hunk 2 above)
Comment (TomKlotzPro) [reply tests/Integration/SettingsTest.php]: done

File: tests/Integration/SettingsTest.php [hunk 4]
```diff
@@ -19,18 +61,48 @@ protected function setUp()
...
+        /* Set the settings with the following parameters with setSettings */
+        $responses[] = $index->setSettings(array('typoTolerance' => 'min', 'ignorePlurals' => array('en', 'fr'), 'removeStopWords' => array('en', 'fr'), 'distinct' => true));
+
+        /* Wait all collected task to terminate */
+        foreach ($responses as $r) {
+            $r->wait();
+        }
+
+        /*  Get the settings with getSettings after update */
+        $this->settings['typoTolerance'] = true;
```
[the remark this answers is not in the document]
Comment (TomKlotzPro) [reply tests/Integration/SettingsTest.php]: done

**Comment by TomKlotzPro** [comment]
I need a review on my last commit @nunomaduro

**Pull request merged**